src/models/baseline/baseline_model.py

```diff
@@ -101,29 +56,67 @@ def fit(self, dataloader: DataLoader):
 
     args = parser.parse_args()
 
+    # CUDA for PyTorch
+    use_cuda = torch.cuda.is_available()
+    device = torch.device("cuda:0" if use_cuda else "cpu")
+
     # data
-    dm = BaselineDaVinciDataModule(args.data_dir)
+    dm = BaselineDaVinciDataModule(args.data_dir, num_workers=20, batch_size=32)
     dm.setup()
-    print("dm setup")
+    val_dataloader = dm.val_dataloader()
+    sample_img = next(iter(val_dataloader))[0][0]
 
     # sanity check
     print("size of trainset:", len(dm.train_samples))
     print("size of validset:", len(dm.val_samples))
     print("size of testset:", len(dm.test_samples))
 
-    img, target = next(iter(dm.val_dataloader()))
-    print(img.shape)
-    print(target.shape)
-
     # model
-    model = BaselineModel()
-    print("model instance created")
-
-    # train
-    min_mae = model.fit(dm.val_dataloader())
-    global_disparity = model.global_disparity
-
+    model = BaselineModel().to(device)
+    model.eval()
+
+    loss_module_dict = {}
+    loss_result_dict = defaultdict(lambda: defaultdict(float))
+    losses_registry = LossRegistry.get_registry()
```

Comment:
Example to get the loss registry. The registry is a dict mapping the loss name (string) to the class of the loss. This allows us to easily add new losses without modifying the baseline code.